### PR TITLE
Small const correctness fix for InputParameters

### LINKED ISSUE #22043: Need to fix a small const correctness issue in InputParameters
## Reason
`isControllable` and `getControllableExecuteOnTypes` can be made as const functions. Without `isControllable` being a const, I cannot call it only with a const input parameters.

## Design
Add const for those two functions to fix their const correctness.

## Impact
More consistent codes and should not have any impact.

### PR DESCRIPTION
**Pull request by YaqiWang**
Close #22043.

### CODE REVIEW

File: framework/src/utils/InputParameters.C [hunk 1]
```diff
@@ -388,15 +388,15 @@ InputParameters::declareControllable(const std::string & input_names,
...
-  return _params.count(name) > 0 && _params[name]._controllable;
+  return _params.count(name) > 0 && _params.at(name)._controllable;
 }
 
 const std::set<ExecFlagType> &
-InputParameters::getControllableExecuteOnTypes(const std::string & name)
+InputParameters::getControllableExecuteOnTypes(const std::string & name) const
 {
-  return _params[name]._controllable_flags;
+  return _params.at(name)._controllable_flags;
```
Comment (loganharbour) [inline framework/src/utils/InputParameters.C:399]:
```suggestion
  const auto it = _params.find(name);
  if (it != _params.end())
    return *it;
  const static std::set<ExecFlagType> empty_types;
  return empty_types;
```

either this or an error. the unintentional insertion we had here before is very bad :(

**Review by loganharbour — CHANGES_REQUESTED** [review]

File: framework/src/utils/InputParameters.C (hunk 1 above)
Comment (YaqiWang) [reply framework/src/utils/InputParameters.C:399]: Yeah, I was hoping a comment like this ;-)

Comment (YaqiWang) [reply framework/src/utils/InputParameters.C:399]: I think it is possibly better to throw an error if the key is not found.

Comment (loganharbour) [reply framework/src/utils/InputParameters.C:399]: If so, then we should probably do the same for `isControllable`

Comment (YaqiWang) [reply framework/src/utils/InputParameters.C]: That has the check whether the parameter is valid. Hold on, actually `Metadata` has a special implementation of `at`, so my original code is ok.

**Comment by YaqiWang** [comment]
@loganharbour this should be ready.

**Review by loganharbour — APPROVED** [review]

**Pull request merged**